rfcs/006-typescript.md

```diff
@@ -0,0 +1,55 @@
+# RFC 005 - Use Typescript on the API
+
+## Affected projects
+
+- API
+
+## Motivation
+
+As discussed during the Brussels team retreat, using Typescript on the API could bring
+interesting benefits such as:
+
+- Better error detection
+
+[It is said](http://ttendency.cs.ucl.ac.uk/projects/type_study/documents/type_study.pdf) that
+types can help to prevent ~15% bugs. They are especialy useful to specify what's nullable,
+what's not and the expected return types. Issues like https://github.com/opencollective/opencollective/issues/1596
+could have been prevented by using Typescript.
+
+- Cleaner patterns
+
+By providing `interface` and `enum` types out of the box, Typescript makes the code cleaner
+and easier to understand in places that use them.
+
+- Developer expenrience
```

Comment:
```suggestion
- Developer experience
```

Reply:
Thanks, I need to add a typo checker to my IDE...